### DISCUSSION TITLE
Playback Issue

### DISCUSSION THREAD

**Discussion by mohankumar-dev-2002** [bluenviron/mediamtx#3045] [Questions]
### Question

Mediamtx recording streams in the format of MPEG, The recording cannot played in either vlc or the browser, One more thing one Firefox browser can load the MP4 format and play it with controls but Chrome or vlc cannot control the playback. I am new to this project any guidance through it will be appreciatable.

**Comment by Taps7734** [comment]
I mentioned this in https://github.com/bluenviron/mediamtx/discussions/2831 as well.  My workaround is to run all the recorded MP4 files through ffmpeg and it seems to fix the videos.

ffmpeg -i inputfile.mp4 -c copy outputfile.mp4

**Comment by mohankumar-dev-2002** [comment]
Hi Taps7734,

I hope you're doing well. I'm reaching out for some help with the
runOnRecordSegmentComplete: hook. I'm thinking of using it to copy the
completed segment to the same file for a smoother process. Since I'm new to
this, I'd really appreciate your guidance. Would you be available for a
quick virtual call to walk me through it?

Thanks,
Mohan Kumar M

**Comment by aler9** [comment]
Hello, regarding recordings in MPEG-TS, browsers do not support MPEG-TS so we cannot do anything about it. Regarding playing-back MPEG-TS recordings with the playback server, i've created an enhancement request here: #3054

**Comment by czullo85** [comment]
Hi all,
I've investigated the issue we're encountering with video stream playback in Chrome compared to Firefox. It appears to be related to how the two browsers handle HTTP requests for media files, particularly in relation to support for "Range Requests" and server responses.
Differences in Browser Behavior:
Chrome: To support functionalities such as seeking, Chrome expects the server to respond to range requests with an HTTP 206 Partial Content status and to include the Content-Range and Content-Length headers in the response. This approach is compliant with HTTP specifications and enables Chrome to understand which parts of the file have been received and which are still needed. It facilitates features like precise seeking and progressive loading.
Firefox: Although it prefers a response similar to Chrome's to optimize seeking operations, Firefox is more tolerant and can play the media file even if the server responds with a 200 OK status for range requests, instead of 206 Partial Content. This tolerance allows for greater flexibility in some server configurations but may also lead to less optimal management of seeking and buffering.
For confirmation, I served the same file using both MediaMtx and Nginx. MediaMtx responds with "200 OK", and seeking does not work in Chrome. In contrast, with Nginx, which replies with "206 Partial Content", it works fine.

**Reply by aler9** [reply]
Hello, in MediaMTX, video returned by the playback server is generated on-demand, therefore range requests are not supported, since they work by requesting a byte range of a file that is not static but generated every time by using existing recordings.

If you want to implement seeking, you can implement a on-click handler on the seeking bar that allows to change the starting timestamp of the playback.

If you're having troubles with the format of videos returned by the playback server, you can also try out the new "standard MP4" format, that is available here: https://github.com/bluenviron/mediamtx/pull/3213

[a comment by an automated account is not shown]
